language/move-vm/integration-tests/src/tests/mutated_accounts_tests.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::compiler::{as_module, compile_units};
+use move_core_types::{
+    account_address::AccountAddress,
+    gas_schedule::{GasAlgebra, GasUnits},
+    identifier::Identifier,
+    language_storage::ModuleId,
+};
+use move_vm_runtime::{data_cache::TransactionEffects, logging::NoContextLog, move_vm::MoveVM};
+use move_vm_test_utils::{ChangeSet, InMemoryStorage};
+use move_vm_types::{
+    gas_schedule::{zero_cost_schedule, CostStrategy},
+    values::Value,
+};
+
+const TEST_ADDR: AccountAddress = AccountAddress::new([42; AccountAddress::LENGTH]);
+
+fn txn_effect_to_changeset(effect: TransactionEffects) -> ChangeSet {
+    let mut changes = ChangeSet::new();
+
+    for (addr, ops) in effect.resources {
+        for (ty, value) in ops {
+            match value {
+                Some((layout, value)) => {
+                    changes.publish_resource(addr, ty, value.simple_serialize(&layout).unwrap())
+                }
+                None => changes.unpublish_resource(addr, ty),
+            }
+            .unwrap();
+        }
+    }
+    changes
+}
```

Comment:
There's already a similar function at `move-vm/test-utils/src/misc.rs`.

Reply:
In the revised version of this PR in #6918, I changed to use the existing function.